pallets/subtensor/src/subnets/mod.rs

```diff
@@ -1,8 +1,39 @@
 use super::*;
+use codec::{Decode, Encode, MaxEncodedLen};
 pub mod lock;
 pub mod registration;
 pub mod serving;
 pub mod subnet;
 pub mod tempo;
 pub mod uids;
 pub mod weights;
+
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    Decode,
+    Default,
+    Encode,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    MaxEncodedLen,
+    scale_info::TypeInfo,
+)]
+pub enum Mechanism {
```

Comment:
maybe also `repr(u8)` so we can hint to the compiler that we are never going to add many many more variants?